quick-protobuf/examples/pb_rs/data_types.rs

```diff
@@ -164,71 +209,74 @@ impl<'a> MessageRead<'a> for FooMessage<'a> {
 impl<'a> MessageWrite for FooMessage<'a> {
     fn get_size(&self) -> usize {
         0
-        + self.f_int32.as_ref().map_or(0, |m| 1 + sizeof_varint(*(m) as u64))
-        + self.f_int64.as_ref().map_or(0, |m| 1 + sizeof_varint(*(m) as u64))
-        + self.f_uint32.as_ref().map_or(0, |m| 1 + sizeof_varint(*(m) as u64))
-        + self.f_uint64.as_ref().map_or(0, |m| 1 + sizeof_varint(*(m) as u64))
-        + self.f_sint32.as_ref().map_or(0, |m| 1 + sizeof_sint32(*(m)))
-        + if self.f_sint64 == 4i64 { 0 } else { 1 + sizeof_sint64(*(&self.f_sint64)) }
-        + if self.f_bool == true { 0 } else { 1 + sizeof_varint(*(&self.f_bool) as u64) }
-        + self.f_FooEnum.as_ref().map_or(0, |m| 1 + sizeof_varint(*(m) as u64))
-        + self.f_fixed64.as_ref().map_or(0, |_| 1 + 8)
-        + self.f_sfixed64.as_ref().map_or(0, |_| 1 + 8)
-        + if self.f_fixed32 == 0u32 { 0 } else { 1 + 4 }
-        + self.f_sfixed32.as_ref().map_or(0, |_| 1 + 4)
-        + self.f_double.as_ref().map_or(0, |_| 1 + 8)
-        + self.f_float.as_ref().map_or(0, |_| 1 + 4)
-        + self.f_bytes.as_ref().map_or(0, |m| 1 + sizeof_len((m).len()))
-        + self.f_string.as_ref().map_or(0, |m| 2 + sizeof_len((m).len()))
-        + self.f_self_message.as_ref().map_or(0, |m| 2 + sizeof_len((m).get_size()))
-        + self.f_bar_message.as_ref().map_or(0, |m| 2 + sizeof_len((m).get_size()))
-        + self.f_repeated_int32.iter().map(|s| 2 + sizeof_varint(*(s) as u64)).sum::<usize>()
-        + if self.f_repeated_packed_int32.is_empty() { 0 } else { 2 + sizeof_len(self.f_repeated_packed_int32.iter().map(|s| sizeof_varint(*(s) as u64)).sum::<usize>()) }
+        + if self.f_int32.is_some() { 1 + sizeof_varint((*(self.f_int32.as_ref().unwrap())) as u64) } else { 0 }
```

Comment:
The old code (using `map_or`) seems more idiomatic to me. Why change to this?

Reply:
Point taken; have changed this back to `map_or()`

The change away was originally just a side effect of me trying to make the codegen less fiddly for maintenance purposes -- have since realized I can implement `map_or()` without too much mess.